app/models/katello/glue/pulp/repo.rb

```diff
@@ -215,11 +215,11 @@ def importer_ssl_options(capsule = SmartProxy.default_capsule!)
             :ssl_client_key => ueber_cert[:key],
             :ssl_ca_cert => ::Cert::Certs.ca_cert
           }
-        elsif self.try(:redhat?) && self.content_view.default?
+        elsif self.try(:redhat?) && self.content_view.default? && Katello::Resources::CDN::CdnResource.redhat_cdn?(url)
           importer_options = {
             :ssl_client_cert => self.product.certificate,
             :ssl_client_key => self.product.key,
-            :ssl_ca_cert => Resources::CDN::CdnResource.ca_file_contents
+            :ssl_ca_cert => Katello::Repository.feed_ca_cert(url)
           }
         else
```

Comment:
When using a redhat url, you are a) providing a client cert and key and b) specifying a particular ca certificate

When using a non-redhat url (https or http), you are not using a client cert or key, and you want to tell pulp to use the system-wide ca trust store, so here we do that by not specifying a ca cert.  Does that answer your question?